### DISCUSSION TITLE
python setting up..

### DISCUSSION THREAD

**Discussion by sachinchaudhary1808** [nix-community/nixvim#1123] [Q&A]
hey i'm new to python can some one suggest me plugins to add ?

and how can i add black python formatted in python 

i'm using nixvim with home manager as a nix module on 23.11 channel

**Comment by martenlindblad** [comment]
```
      plugins = {
        lsp = {
          enable = true;
          servers = {
            ruff-lsp.enable = true;
          };
        };
        conform-nvim = {
          enable = true;
          formatOnSave = {
            lspFallback = true;
            timeoutMs = 1500;
          };
          formattersByFt = {
            python = ["black"];
          };
        };
      };

```

**Reply by sachinchaudhary1808** [reply]
thank you sir it's pretty old comment but this is how my config looks like now"

`{
  pkgs,
  lib,
  ...
}: let
  nixvim = import (builtins.fetchGit {
    url = "https://github.com/nix-community/nixvim";
    # If you are not running an unstable channel of nixpkgs, select the corresponding branch of nixvim.
    ref = "nixos-23.11";
  });
in {
  imports = [
    # For home-manager
    # nixvim.homeManagerModules.nixvim
    # For NixOS
    nixvim.nixosModules.nixvim
    # For nix-darwin
    # nixvim.nixDarwinModules.nixvim
  ];

  programs.nixvim = {
    enable = true;
    colorschemes.gruvbox.enable = true;
    #colorschemes.catppuccin = {
    #     enable = true;
    #    flavour = "mocha";
    # transparentBackground = true;
    #    };

    options = {
      number = true; # Show line numbers
      relativenumber = true; # Show relative line numbers

      shiftwidth = 2; # Tab width should be 2
      clipboard = "unnamedplus";
    };
  };

  programs.nixvim = {
    plugins.lualine.enable = true;
   # plugins.commentary.enable = true;
    plugins.telescope.enable = true;
    plugins.comment-nvim.enable = true;
    plugins.neo-tree.enable = true;
   # plugins.twilight.enable = true;
    plugins.lsp.servers.pyright.enable = true;
    plugins.nvim-cmp.enable = true;
    plugins.treesitter.enable = true;
    plugins.lsp.enable = true;
    plugins.nvim-cmp = {
      sources = [
        {name = "nvim_lsp";}
      ];
    };

    plugins.nvim-autopairs = {
      enable = true;
    };
  };

  programs.nixvim = {
    keymaps = [
      {
        action = "<cmd>Neotree toggle<CR>";
        key = "<C-n>"; # this line is changed
        mode = "n";
        options = {
          desc = "Toggle Tree View.";
        };
      }
      {
        action = "<cmd>lua vim.lsp.buf.hover()<CR>";
        key = "<C-k>"; # this line is changed
        mode = "n";
        options = {
          desc = "info about things";
        };
      }
      {
        action = "<cmd>term python3 %<CR>";
        key = "<F8>"; # this line is changed
        mode = ["n" "i"];
        options = {
          desc = "runs currunt python file in the terminal";
        };
      }
    ];
    # ...
  };

  programs.nixvim = {
    extraPlugins = with pkgs.vimPlugins; [
      neoformat
    ];
  };

  programs.nixvim.autoCmd = [
    {
      event = ["TermOpen"];
      pattern = ["*"];
      command = "startinsert";
    }
  ];
}`

**Comment by sachinchaudhary1808** [comment]
thanks
